flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/operator/AbstractCEPPatternOperator.java

```diff
@@ -126,11 +126,12 @@ public void restoreState(FSDataInputStream state) throws Exception {
 
 		int numberPriorityQueueEntries = ois.readInt();
 
-		priorityQueue = new PriorityQueue<StreamRecord<IN>>(numberPriorityQueueEntries, new StreamRecordComparator<IN>());
-
-		for (int i = 0; i <numberPriorityQueueEntries; i++) {
-			StreamElement streamElement = streamRecordSerializer.deserialize(new DataInputViewStreamWrapper(ois));
-			priorityQueue.offer(streamElement.<IN>asRecord());
+		if(numberPriorityQueueEntries > 0) {
+			priorityQueue = new PriorityQueue<StreamRecord<IN>>(numberPriorityQueueEntries, new StreamRecordComparator<IN>());
```

Comment:
I think if number of objects read are 0 then why would we want to create object at all? Thats unnecessary and condition check will save some CPU cycles.